lib/Basics/ScopedTimer.h

```diff
@@ -0,0 +1,192 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2017 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Christoph Uhde
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+#ifndef ARANGODB_BASICS_SCOPEDTIMER__H
+#define ARANGODB_BASICS_SCOPEDTIMER__H 1
+
+#include <lib/Logger/Logger.h>
+
+#include <utility>
+#include <vector>
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+namespace arangodb {
+
+
+class ScopedTimer {
+public:  // defines
+    using hclock    = std::chrono::high_resolution_clock;
+    using hrcStrVec = std::vector<std::pair<hclock::time_point,std::string>>;
+    using intStrVec = std::vector<std::pair<int,std::string>>;
+
+private:  // variables
+    hrcStrVec _timePoints;
+    std::function<void(intStrVec)> _callback;
+    bool _enabled;
+    bool _add_dtor_entry;
```

Comment:
note that we use camelCase and no snake_case as part of our coding conventions.

Reply:
I forgot to change every variable when i imported the code form my repo. I'll fix the internal stuff as well.